File: app/src/main/java/br/edu/ifspsaocarlos/agenda/activity/SearchActivity.java
package br.edu.ifspsaocarlos.agenda.activity;

import android.app.SearchManager;
import android.content.Intent;
import android.support.design.widget.FloatingActionButton;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;

import br.edu.ifspsaocarlos.agenda.R;

public class SearchActivity extends BaseActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        //setContentView(R.layout.activity_main);

        FloatingActionButton myFab = (FloatingActionButton) findViewById(R.id.fab);
        myFab.hide();

        Intent intent = getIntent();
        if (Intent.ACTION_SEARCH.equals(intent.getAction())) {
            String query = intent.getStringExtra(SearchManager.QUERY);
            buildSearchListView(query);
        }



    }
}

File: app/src/main/java/br/edu/ifspsaocarlos/agenda/activity/SearchActivity.kt
package br.edu.ifspsaocarlos.agenda.activity

import android.app.SearchManager
import android.content.Intent
import android.os.Bundle
import br.edu.ifspsaocarlos.agenda.R
import com.google.android.material.floatingactionbutton.FloatingActionButton

class SearchActivity : BaseActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        val myFab = findViewById<FloatingActionButton>(R.id.fab)
        myFab.hide()

        val intent = intent
        handleIntent(intent)
    }

    override fun onNewIntent(intent: Intent) {
        super.onNewIntent(intent)
        setIntent(intent)
        handleIntent(intent)
    }

    private fun handleIntent(intent: Intent) {
        if (Intent.ACTION_SEARCH == intent.action) {
            val query = intent.getStringExtra(SearchManager.QUERY) ?: ""
            buildSearchListView(query)
        }
    }
}